Tools/check-c-api-docs/main.py

```diff
@@ -0,0 +1,166 @@
+import re
+from pathlib import Path
+import sys
+import _colorize
+
+SIMPLE_FUNCTION_REGEX = re.compile(r"PyAPI_FUNC(.+) (\w+)\(")
+SIMPLE_MACRO_REGEX = re.compile(r"# *define *(\w+)(\(.+\))? ")
+SIMPLE_INLINE_REGEX = re.compile(r"static inline .+( |\n)(\w+)")
+SIMPLE_DATA_REGEX = re.compile(r"PyAPI_DATA\(.+\) (\w+)")
+
+MISTAKE = """\
+If this is a mistake and this script should not be failing, please create an
+issue and tag Peter (@ZeroIntensity) on it.\
+"""
+
+FOUND_UNDOCUMENTED = f"""\
+Found some undocumented C API(s)!
+
+Python requires documentation on all public C API symbols, macros, and types.
+If these API(s) were not meant to be public, please prefix them with a
+leading underscore (_PySomething_API) or move them to the internal C API
+(pycore_*.h files).
+
+In exceptional cases, certain functions can be ignored by adding them to
+Tools/c-api-docs-check/ignored_c_api.txt
+
+{MISTAKE}\
+"""
+
+FOUND_IGNORED_DOCUMENTED = f"""\
+Some C API(s) were listed in Tools/c-api-docs-check/ignored_c_api.txt, but
+they were found in the documentation. To fix this, simply update ignored_c_api.txt
+accordingly.
```

Comment:
```suggestion
they were found in the documentation. To fix this, update ignored_c_api.txt
accordingly.
```
Or tell directly what to do?
```suggestion
they were found in the documentation. To fix this, remove them from
ignored_c_api.txt.
```